healthcare/api-client/v1/hl7v2/hl7v2_messages_test.py

```diff
@@ -0,0 +1,191 @@
+import datasets
+import hl7v2_stores
+import hl7v2_messages
+
+cloud_region = 'us-central1'
+project_id = os.environ['GOOGLE_CLOUD_PROJECT']
+
+dataset_id = 'test_dataset_{}'.format(uuid.uuid4())
+hl7v2_store_id = 'test_hl7v2_store-{}'.format(uuid.uuid4())
+hl7v2_message_file = 'resources/hl7-sample-ingest.json'
+label_key = 'PROCESSED'
+label_value = 'TRUE'
+
+
+@pytest.fixture(scope='module')
+def test_dataset():
+    dataset = datasets.create_dataset(
+        project_id,
+        cloud_region,
+        dataset_id)
+
+    yield dataset
+
+    # Clean up
+    datasets.delete_dataset(
+        project_id,
+        cloud_region,
+        dataset_id)
+
+
+@pytest.fixture(scope='module')
+def test_hl7v2_store():
+    hl7v2_store = hl7v2_stores.create_hl7v2_store(
+        project_id,
+        cloud_region,
+        dataset_id,
+        hl7v2_store_id)
+
+    yield hl7v2_store
+
+    hl7v2_stores.delete_hl7v2_store(
+        project_id,
+        cloud_region,
+        dataset_id,
+        hl7v2_store_id)
+
+
+def test_CRUD_hl7v2_message(test_dataset, test_hl7v2_store, capsys):
+    hl7v2_messages.create_hl7v2_message(
+        project_id,
+        cloud_region,
+        dataset_id,
+        hl7v2_store_id,
+        hl7v2_message_file)
+
+        hl7v2_messages_list = hl7v2_messages.list_hl7v2_messages(
+            project_id,
+            cloud_region,
+            dataset_id,
+            hl7v2_store_id)
+
+        assert len(hl7v2_messages_list) > 0
+        hl7v2_message_name = hl7v2_messages_list[0].get('name')
+        nonlocal hl7v2_message_id
+        hl7v2_message_id = hl7v2_message_name.split('/', 9)[9]
+
+    hl7v2_messages.get_hl7v2_message(
+        project_id,
+        cloud_region,
+        dataset_id,
+        hl7v2_store_id,
+        hl7v2_message_id)
+
+    hl7v2_messages.delete_hl7v2_message(
+        project_id,
+        cloud_region,
+        dataset_id,
+        hl7v2_store_id,
+        hl7v2_message_id)
+
+    out, _ = capsys.readouterr()
+
+    # Check that create/get/list/delete worked
+    assert 'Created HL7v2 message' in out
+    assert 'Name' in out
+    assert 'Deleted HL7v2 message' in out
+
+
+def test_ingest_hl7v2_message(test_dataset, test_hl7v2_store, capsys):
+    hl7v2_messages.ingest_hl7v2_message(
+        project_id,
+        cloud_region,
+        dataset_id,
+        hl7v2_store_id,
+        hl7v2_message_file)
+
+        hl7v2_messages_list = hl7v2_messages.list_hl7v2_messages(
+            project_id,
+            cloud_region,
+            dataset_id,
+            hl7v2_store_id)
+
+        assert len(hl7v2_messages_list) > 0
+        hl7v2_message_name = hl7v2_messages_list[0].get('name')
+        nonlocal hl7v2_message_id
+        hl7v2_message_id = hl7v2_message_name.split('/', 9)[9]
+
+    hl7v2_messages.get_hl7v2_message(
+        project_id,
+        cloud_region,
+        dataset_id,
+        hl7v2_store_id,
+        hl7v2_message_id)
+
+    hl7v2_messages.delete_hl7v2_message(
```

Comment:
If this delete call failed, what happens when the fixture tries to delete the store or dataset? If the API allows you to delete the store which is non-empty, this seems fine.

Reply:
Yes, you can delete a non-empty store.